partiql-grammar/src/main/antlr/org/partiql/grammar/PartiQL.g4

```diff
@@ -539,7 +553,7 @@ extract
     : EXTRACT PAREN_LEFT IDENTIFIER FROM rhs=expr PAREN_RIGHT;
 
 trimFunction
-    : func=TRIM PAREN_LEFT ( mod=IDENTIFIER? sub=expr? FROM )? target=expr PAREN_RIGHT;
+    : func=TRIM PAREN_LEFT ( mod=(BOTH|LEADING|TRAILING)? sub=expr? FROM )? target=expr PAREN_RIGHT;
```

Comment:
Is pattern restrictor its own AST node? Also I was just reading through the grammar and noticed that we don't need many of the element labels. The element and list element labels are useful when the rule has more than one element with the same type. Unless that's the case, they're clutter

For example, we don't need any of the element labels here. It makes it harder to read
Ex: https://github.com/partiql/partiql-lang-kotlin/blob/main/partiql-grammar/src/main/antlr/org/partiql/grammar/PartiQL.g4#L329-L337

Reply:
For this example, are you talking about labels `sub` and `target`? Only did that because `sub` is optional.

Or `func`? I added the `func` label just because it made the PartiQLVisitor code look uniform. AKA DateFunction, and TrimFunction, and others result in:

```kotlin
call(ctx.func.text.toLowerCase(), args, metas)
```

But doesn't matter to me.

And yes, PatternRestrictor is its own AST node.